### PR TITLE
Fix inspector module export function

### PR DESCRIPTION
**Pull request by vzhestkov**
### What does this PR do?

Fixes various tracebacks on calling `inspector.export`.

### Previous Behavior
Different tracebacks on calling `inspector.export`

### New Behavior
Normal function behavior

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### CODE REVIEW

**Comment by garethgreenaway** [comment]
@vzhestkov This one needs a changelog and some tests.  Thanks!

**Comment by vzhestkov** [comment]
@garethgreenaway I've added the changelog entry, but for the tests I see that very limited amount of `inspectlib` is covered with the tests. Where should I put the test for missing `type` for the block device returned by `salt.utils.fsutils._blkid()` and how to properly mock `gzip.open` to perform a real tests, or checking the `mode` passed to `gzip.open` is enough in this case?

**Comment by garethgreenaway** [comment]
@vzhestkov Apologies for the delay in responding.   I think just testing that the right mode is being passed into the gzip.open mock is enough.  We just want to make sure it doesn't get flipped back to the previous mode and if it does inadvertently then these tests will fail.  Something simple like this:

```
diff --git a/tests/unit/modules/inspectlib/test_fsdb.py b/tests/unit/modules/inspectlib/test_fsdb.py
index f5a0c2bdf1..6d249e665c 100644
--- a/tests/unit/modules/inspectlib/test_fsdb.py
+++ b/tests/unit/modules/inspectlib/test_fsdb.py
@@ -126,15 +126,19 @@ class InspectorFSDBTestCase(TestCase):
             "os.listdir", MagicMock(return_value=["some_table"])
         ):
             writable = Writable()
-            with patch("gzip.open", MagicMock(return_value=writable)):
+            with patch("gzip.open", MagicMock(return_value=writable)) as gzip_mock_open:
                 csvdb = CsvDB("/foobar")
                 csvdb.open()
                 csvdb.create_table_from_object(FoobarEntity())
 
+            # test the second call to gzip.open, the first is in the list_tables function
+            assert gzip_mock_open.call_args_list[1][0][1] == 'wt'
+
             sorted_writable_data = sorted(writable.data[0].strip().split(","))
             sorted_expected_data = sorted("foo:int,bar:str,spam:float".split(","))
             self.assertEqual(sorted_writable_data, sorted_expected_data)
 
+
     def test_list_databases(self):
         """
         Test list databases.
@@ -153,7 +157,7 @@ class InspectorFSDBTestCase(TestCase):
             "os.listdir", MagicMock(return_value=["some_table"])
         ):
             writable = Writable()
-            with patch("gzip.open", MagicMock(return_value=writable)):
+            with patch("gzip.open", MagicMock(return_value=writable)) as gzip_mock_open:
                 obj = FoobarEntity()
                 obj.foo = 123
                 obj.bar = "test entity"
@@ -170,6 +174,10 @@ class InspectorFSDBTestCase(TestCase):
                     )
                 }
                 csvdb.store(obj)
+
+                # test the second call to gzip.open, the first is in the list_tables function
+                assert gzip_mock_open.call_args_list[1][0][1] == 'at'
+
                 assert writable.data[0].strip() == "123,test entity,0.123"
 
     def test_delete_object(self):
@@ -269,7 +277,7 @@ class InspectorFSDBTestCase(TestCase):
         """
         with patch("os.listdir", MagicMock(return_value=["test_db"])), patch(
             "gzip.open", MagicMock()
-        ), patch(
+        ) as gzip_mock_open, patch(
             "csv.reader",
             MagicMock(
                 return_value=iter(
@@ -285,6 +293,10 @@ class InspectorFSDBTestCase(TestCase):
             csvdb = CsvDB("/foobar")
             csvdb.open()
             entities = csvdb.get(FoobarEntity)
+
+            # test the second call to gzip.open, the first is in the open function
+            assert gzip_mock_open.call_args_list[1][0][1] == 'rt'
+
             assert list == type(entities)
             assert len(entities) == 2
 
```

For the other change, I'm fine with it without an additional test.  @Ch3LL thoughts?

**Comment by vzhestkov** [comment]
Thanks, @garethgreenaway. I added the tests as you suggested.

**Review by garethgreenaway — DISMISSED** [review]

File: changelog/61530.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix tracabacks caused by missing block device type and wrong mode used for gzip.open while calling inspector.export
```
Comment (Ch3LL) [inline changelog/61530.fixed:1]:
```suggestion
Fix tracebacks caused by missing block device type and wrong mode used for gzip.open while calling inspector.export
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: changelog/61530.fixed (hunk 1 above)
Comment (vzhestkov) [reply changelog/61530.fixed:1]: Ouch, thanks for pointing :+1:

**Review by garethgreenaway — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**